app/controllers/missions_controller.rb

```diff
@@ -27,11 +27,15 @@ def show; end
   def edit; end
 
   def update
-    if @mission.update(permitted_params)
+    update_transaction = Missions::UpdateTransaction.new.with_step_args(
+      update: [mission: @mission]
+    ).call(permitted_params)
+
+    if update_transaction.success?
       flash[:notice] = translate 'activerecord.notices.messages.update_success'
       render :show
     else
-      flash[:error] = translate 'activerecord.errors.messages.update_fail'
+      flash[:error] = update_transaction.failure
```

Comment:
Same question as above

Reply:
The message is rendered properly on view